scripts/release_info.py

```diff
@@ -243,14 +245,47 @@ def main():
             for item in storage_changes:
                 out.write('* %s  \n' % item)
 
-        out.write('\n### Authors:\n')
+        past_authors = {
+            log.email: log.author for log in past_logs
+        }
+        release_authors = {(log.author, log.email) for log in new_release_logs}
+
+        new_authors = sorted(set(
+            [(name, email) for name, email in release_authors
+             if email not in past_authors]))
+        existing_authors = sorted(set(
+            [(name, email) for name, email in release_authors
+             if email in past_authors]))
+        new_author_names = [name for name, _ in new_authors]
+        existing_author_names = [name for name, _ in existing_authors]
+
         # TODO: duplicate author handling due to email changes
-        for name, email in sorted({(log.author, log.email) for log in logs}):
-            out.write('%s <%s>  \n' % (name, email))
+        out.write('\n### New Authors:\n')
+        for name, email in new_authors:
+            out.write('* %s <%s>\n' % (name, email))
+
+        out.write('\n### Existing Authors:\n')
+        for name, email in existing_authors:
+            out.write('* %s <%s>\n' % (name, email))
+
+        # Generate the author sections of the email.
+        out.write('\n### Email C&P Blurbs about authors:\n')
+        new_author_comma_list = (
+            '%s, and %s' % (', '.join(
+                new_author_names[:-1]), new_author_names[-1]))
+        existing_author_comma_list = (
+            '%s, and %s' % (', '.join(
+                existing_author_names[:-1]), existing_author_names[-1]))
+        out.write(
+            '``Please welcome %s for whom this release marks their first '
+            'contribution to Oppia!``\n\n' % new_author_comma_list)
+        out.write(
+            '``Thanks to %s, our returning contributors who made this release '
```

Comment:
Just checking, does this only print existing authors who've contributed to this specific release?

Reply:
Yep!